tests/rpminspect/rpminspect.sh

```diff
@@ -50,45 +49,72 @@ rlJournalStart
 		rlRun "tree ./inspect_builds"
 
 		## Do actual rpminspect
-		# Default and required options
-		args="-v -c ${RPMINSPECT_CONFIG:-/usr/share/rpminspect/fedora.yaml}"
-		# Output the data to json so that it can be displayed
-		args="$args --output=$TMT_TEST_DATA/result.json --format=json"
+		rlRun "args=\"-v -c ${RPMINSPECT_CONFIG:-/usr/share/rpminspect/${ID}.yaml}\"" 0 "Set args: Default and required options"
+		rlRun "args=\"\$args --output=$TMT_TEST_DATA/result.json --format=json\"" 0 "Set args: Output the data to json"
 		# Specify the test to run
 		if [[ -n "$RPMINSPECT_TESTS" ]]; then
-			# Run only specified tests. Takes precedence over --exclude
-			args="$args --tests=$RPMINSPECT_TESTS"
+			# Takes precedence over --exclude
+			rlRun "args=\"\$args --tests=$RPMINSPECT_TESTS\"" 0 "Set args: Run only specified tests"
 		elif [[ -n "$RPMINSPECT_EXCLUDE" ]]; then
-			# Exclude test lists given. Only run if there is no RPMINSPECT_TESTS
-		 	args="$args --exclude=${RPMINSPECT_EXCLUDE:-metadata}"
+			# Only run if there is no RPMINSPECT_TESTS
+			rlRun "args=\"\$args --exclude=${RPMINSPECT_EXCLUDE}\"" 0 "Set args: Exclude test lists given"
 		else
 			# TODO: Only exclude metadata if running with copr
 			# https://tmt.readthedocs.io/en/stable/spec/context.html#initiator
-			args="$args --exclude=metadata"
+			rlRun "args=\"\$args --exclude=metadata\"" 0 "Set args: Exclude metadata on copr"
 		fi
-		# Run rpminspect for the specified architectures
-		[[ -n "$RPMINPSECT_ARCHES" ]] && args="$args --arches=$RPMINPSECT_ARCHES"
-		# If we have a previous build to compare with, use that as before_build
-		[[ -n "$latest_build" ]] && args="$args ./inspect_builds/$latest_build"
-		# The remaining part is treated as the after_build/the build to be inspected
-		args="$args ./inspect_builds/$PACKIT_PACKAGE_NVR"
+		[[ -n "$RPMINPSECT_ARCHES" ]] && rlRun "args=\"\$args --arches=$RPMINPSECT_ARCHES\"" 0 "Set args: Run rpminspect for the specified architectures"
+		[[ -n "$latest_build" ]] && rlRun "args=\"\$args ./inspect_builds/$latest_build\"" 0 "Set args: Use latest_build as before_build"
+		rlRun "args=\"\$args --threshold=${RPMINSPECT_THRESHOLD:-BAD}\"" 0 "Set args: Failure threshold"
+		[[ -n "${RPMINSPECT_SUPPRESS}" ]] && rlRun "args=\"\$args --suppress=${RPMINSPECT_SUPPRESS}\"" 0 "Set args: Suppress threshold"
+		rlRun "args=\"\$args ./inspect_builds/$PACKIT_PACKAGE_NVR\" && echo \$args" 0 "Set args: Set downloaded source as after_build"
 	  rlRun "/usr/bin/rpminspect $args" 0 "Run rpminspect"
+		# Get the number of BAD and VERIFY check results
+		# Note about jq syntax used:
+		# - .[][]               : collapse the dict of list of objects into a series of objects
+		# - select(.result ...) : filter the series of objects by the `result` value
+		# - [ ... ] | length    : create a list of the filtered object series and count the length
```

Comment:
Thanks for the comment!

Of topic:
I must say beakerlib is starting to make this hard to read. (I wonder if we can't switch to a different platform/tool/syntax before it's too late..)

Reply:
Syntax wise I agree. But the `journal.txt` and `output.txt` are very neat in debugging such issues (that's why I've added the `args` building in there). Some IDE support would definitely go a long way.